src/recordlinker/schemas/pii.py

```diff
@@ -392,25 +395,37 @@ def blocking_keys(self, key: models.BlockingKey) -> set[str]:
             # NOTE: we could optimize here and remove the dashes from the date
             vals.update(self.feature_iter(Feature(attribute=FeatureAttribute.BIRTHDATE)))
         elif key == models.BlockingKey.IDENTIFIER:
-            vals.update({
-                f"{type_part}:{authority_part[:2]}:{value_part[-4:]}"
-                for x in self.feature_iter(Feature(attribute=FeatureAttribute.IDENTIFIER))
-                for type_part, authority_part, value_part in [x.split(":", 2)]
-            })
+            vals.update(
+                {
+                    f"{value_part[-4:]}:{authority_part[:2]}:{type_part}"
```

Comment:
I think this is fine for consistency, but technically blocking keys don't have the same problem since we are also check for equality when writing our SQL where statements.